private/buf/buffetch/internal/module_ref.go

```diff
@@ -61,6 +61,10 @@ func newDirectModuleRef(format string, moduleReference bufmoduleref.ModuleRefere
 	}
 }
 
+func (r *moduleRef) ID() string {
+	return r.moduleReference.IdentityString()
```

Comment:
This isn't unique necessarily - a module ref may include a branch, tag, or commit associated with it. Optimally, an ID would be the full ID with the commit, but regardless we can't just have the identity.